plugins/linuxplugin/ifplugin/data_resync.go

```diff
@@ -33,21 +58,302 @@ func (plugin *LinuxInterfaceConfigurator) Resync(interfaces []*interfaces.LinuxI
+			if err != nil {
+				plugin.Log.Error(err)
+				continue
+			}
+			if nbIPNet.Mask.String() == vppIPNet.Mask.String() && bytes.Compare(pNbIP, pVppIP) == 0 {
+				ipFound = true
+				break
+			}
+		}
+		if !ipFound {
+			plugin.Log.Debugf("Interface RESYNC comparison: linux interface %v does not contain IP %s", nbIf.Name, nbIP)
+			return true
+		}
+	}
+	// Physical address
+	if nbIf.PhysAddress != linuxIf.PhysAddress {
+		plugin.Log.Debugf("Interface RESYNC comparison: MAC address changed (NB: %s, Linux: %s)",
+			nbIf.PhysAddress, linuxIf.PhysAddress)
+		return true
+	}
+	// MTU
+	if nbIf.Mtu != linuxIf.Mtu {
+		plugin.Log.Debugf("Interface RESYNC comparison: MTU changed (NB: %d, Linux: %d)",
+			nbIf.Mtu, linuxIf.Mtu)
+		return true
+	}
+	switch nbIf.Type {
+	case interfaces.LinuxInterfaces_VETH:
+		if nbIf.Veth == nil && linuxIf.Veth != nil || nbIf.Veth != nil && linuxIf.Veth == nil {
+			plugin.Log.Debugf("Interface RESYNC comparison: VETH setup changed (NB: %v, VPP: %v)",
+				nbIf.Veth, linuxIf.Veth)
+			return true
+		}
+		if nbIf.Veth != nil && linuxIf.Veth != nil {
+			// VETH peer name
+			if nbIf.Veth.PeerIfName != linuxIf.Veth.PeerIfName {
+				plugin.Log.Debugf("Interface RESYNC comparison: VETH peer name changed (NB: %s, VPP: %s)",
+					nbIf.Veth.PeerIfName, linuxIf.Veth.PeerIfName)
+				return true
+			}
+		}
+	case interfaces.LinuxInterfaces_AUTO_TAP:
+		// Host name for TAP
+		if nbIf.HostIfName != linuxIf.HostIfName {
+			plugin.Log.Debugf("Interface RESYNC comparison: TAP host name changed (NB: %d, Linux: %d)",
+				nbIf.HostIfName, linuxIf.HostIfName)
+			return true
+		}
+		// Note: do not compare TAP temporary name. It is local-only parameter which cannot be leveraged externally.
+	}
+
+	return false
+}
+
+// Looks for linux interface. Returns net.Link object if found
+func (plugin *LinuxInterfaceConfigurator) findLinuxInterface(nbIf *interfaces.LinuxInterfaces_Interface, nsMgmtCtx *linuxcalls.NamespaceMgmtCtx) (netlink.Link, error) {
+	plugin.Log.Debugf("Looking for Linux interface %v", nbIf.HostIfName)
+
+	// Move to proper namespace
+	var revertNs func()
+	if nbIf.Namespace != nil {
+		if nbIf.Namespace != nil {
+			if !plugin.isNamespaceAvailable(nbIf.Namespace) {
+				return nil, fmt.Errorf("RESYNC Linux interface %s: namespace is not available", nbIf.HostIfName)
+			}
+			// Switch to namespace
+			var err error
+			revertNs, err = plugin.switchToNamespace(nsMgmtCtx, nbIf.Namespace)
+			if err != nil {
+				return nil, fmt.Errorf("RESYNC Linux interface %s: failed to switch to namespace %s: %v",
+					nbIf.HostIfName, nbIf.Namespace.Name, err)
+			}
+		}
+	}
+
+	// Define defer func to revert namespace if needed
+	defer func(revertNs func()) {
+		if revertNs != nil {
+			revertNs()
+		}
+	}(revertNs)
+
+	// Look for interface
+	linkIf, err := netlink.LinkByName(nbIf.HostIfName)
+	if err != nil {
+		// Link not found is not an error in this case
+		if err.Error() == "Link not found" {
```

Comment:
Seems to me it would be better to compare this error to [LinkNotFoundError](https://godoc.org/github.com/vishvananda/netlink#LinkNotFoundError) type instead of comparing strings.